src/client/httpClient.ts

```diff
@@ -263,68 +363,26 @@ export class HTTPClient {
         }
     }
 
-    makeHTTP2Request(
+    async makeHTTP2Request(
         path: string,
         token: string,
         body: Record<string, any>,
         timeout: number
     ): Promise<{ status: number, data: Record<string, any> }> {
-        return new Promise((resolve, reject) => {
-            // Should never happen, but good to have a readable error just in case
-            if (this.http2Session == null) {
-                throw new Error('Cannot make http2 request without session');
-            }
-            if (this.closed) {
-                throw new Error('Cannot make http2 request when client is closed');
-            }
-
-            // Recreate session if session was closed except via an explicit `close()`
-            // call. This happens when nginx sends a GOAWAY packet after 1000 requests.
-            if (this.http2Session.closed) {
```

Comment:
So, this is not happening it seems? Am I correct?

Reply:
This does happen after you send 1000 requests, but the removed implementation isn't sufficient to handle > 1000 requests in parallel. It worked well with > 1000 requests in series.